ocs_ci/ocs/pgsql.py

```diff
@@ -0,0 +1,293 @@
+        """
+        Get all postgresql pods
+
+        Returns:
+            List: postgresql pod objects list
+
+        """
+        return get_all_pods(
+            namespace=RIPSAW_NAMESPACE, selector=['postgres']
+        )
+
+    def get_pgbench_pods(self):
+        """
+        Get all pgbench pods
+
+        Returns:
+            List: pgbench pod objects list
+
+        """
+        return [
+            get_pod_obj(
+                pod
+            ) for pod in get_pod_name_by_pattern('pgbench', RIPSAW_NAMESPACE)
+        ]
+
+    def is_pgbench_running(self):
+        """
+        Check if pgbench is running
+
+        Returns:
+            bool: True if pgbench is running; False otherwise
+
+        """
+        pod_objs = self.get_pgbench_pods()
+        for pod in pod_objs:
+            if pod.get().get(
+                'status'
+            ).get('containerStatuses')[0].get('state') == 'running':
+                log.info("One or more pgbench pods are in running state")
+                return True
+            else:
+                return False
+            break
+
+    def get_pgbench_status(self, pgbench_pod_name):
+        """
+        Get pgbench status
+
+        Args:
+            pgbench_pod_name (str): Name of the pgbench pod
+
+        Returns:
+            str: state of pgbench pod (running/completed)
+
+        """
+        pod_obj = get_pod_obj(pgbench_pod_name, namespace=RIPSAW_NAMESPACE)
+        status = pod_obj.get().get(
+            'status'
+        ).get('containerStatuses')[0].get('state')
+        if list(status.keys())[0] == 'running':
+            return 'running'
+        elif list(status.keys())[0] == 'terminated':
+            return status['terminated']['reason']
+
+    def wait_for_pgbench_status(self, status, timeout=None):
+        """
+        Wait for pgbench benchmark pods status to reach running/completed
+
+        Args:
+            status (str): status to reach Running or Completed
+            timeout (int): Time in seconds to wait
+
+        """
+
+        timeout = timeout if timeout else 43200
+        # Wait for pg_bench pods to initialized and running
+        log.info(f"Waiting for pgbench pods to be reach {status} state")
+        pgbench_pod_objs = self.get_pgbench_pods()
+        for pgbench_pod_obj in pgbench_pod_objs:
+            wait_for_resource_state(
+                resource=pgbench_pod_obj, state=status, timeout=timeout
+            )
+
+    def validate_pgbench_run(self, pgbench_pods):
+        """
+        Validate pgbench run
+
+        Args:
+            pgbench pods (list): List of pgbench pods
+
+        Returns:
+            pg_output (list): pgbench outputs in list
+
+        """
+        pg_output = []
+        for pgbench_pod in pgbench_pods:
+            log.info(f"pgbench_client_pod===={pgbench_pod.name}====")
+            output = run_cmd(f'oc logs {pgbench_pod.name}')
+            pg_output = utils.parse_pgsql_logs(output)
```

Comment:
how big is the logs output? If too big, consider logging as debug the full log and as info some highlights, in order to avoid log flooding

Reply:
We are not dumping the entire pgbench run output. parsing the important info from the output and displaying.
snip:
18:49:06 - MainThread - ocs_ci.ocs.pgsql - INFO - pgbench_client_pod====pgbench-1-dbs-client-1-7af87ecc-g5xmh====
18:49:06 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc logs pgbench-1-dbs-client-1-7af87ecc-g5xmh
18:49:07 - MainThread - ocs_ci.ocs.pgsql - INFO - *******PGBench output log*********
[{1: {'scaling_factor': '2', 'num_clients': '20', 'num_threads': '2', 'number_of_transactions_per_client': '100', 'number_of_transactions_actually_processed': '2000', 'latency_avg': '86', 'lat_stddev': '91', 'tps_incl': '217', 'tps_excl': '217'}}, {2: {'scaling_factor': '2', 'num_clients': '20', 'num_threads': '2', 'number_of_transactions_per_client': '100', 'number_of_transactions_actually_processed': '2000', 'latency_avg': '89', 'lat_stddev': '96', 'tps_incl': '212', 'tps_excl': '212'}}, {3: {'scaling_factor': '2', 'num_clients': '20', 'num_threads': '2', 'number_of_transactions_per_client': '100', 'number_of_transactions_actually_processed': '2000', 'latency_avg': '101', 'lat_stddev': '115', 'tps_incl': '190', 'tps_excl': '190'}}, {4: {'scaling_factor': '2', 'num_clients': '20', 'num_threads': '2', 'number_of_transactions_per_client': '100', 'number_of_transactions_actually_processed': '2000', 'latency_avg': '98', 'lat_stddev': '123', 'tps_incl': '191', 'tps_excl': '191'}}]